### PR TITLE
Testing: Make hdf5 test able to run in parallel and guard BP engine tests

### LINKED ISSUE #262: ADIOS1 dependency forced in current master
The current master tries to compile tests that depend on ADIOS1 even though the dependency was not found by cmake

### PR DESCRIPTION
**Pull request by HaochengLIU**
In this PR, I extend HDF5 native reader so that now hdf5 tests can read data
by selection(so called hyperslab selection in hdf5).
Close #262

### CODE REVIEW

**Comment by HaochengLIU** [comment]
@chuckatkins 
Hello Chuck, would you please kindly review this PR? I extend hdf5 native reader so that hdf5 test is parallel compatible!

**Comment by chuckatkins** [comment]
@HaochengLIU Please rebase on master to get updated CI information

File: testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp [hunk 1]
```diff
@@ -187,44 +222,77 @@ void HDF5NativeReader::ReadVar(const std::string varName, void *dataArray)
...
             auto &var_r64 =
-                io.DefineVariable<double>("r64", {}, {}, adios2::Dims{8});
+                io.DefineVariable<double>("r64", shape, start, count);
         }
 
-        // Create the HDF5 Engine
+        // Create the ADIOS 1 Engine
         io.SetEngine("HDF5Writer");
 
+#ifdef ADIOS2_HAVE_MPI
```
Comment (chuckatkins) [inline testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp]: @guj @williamfgc Does the HDF5 engine actually use any transports or is the engine making calls directly to hdf5?

**Comment by HaochengLIU** [comment]
Done.

File: testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp [hunk 2]
```diff
@@ -605,48 +751,85 @@ TEST_F(HDF5WriteReadTest, DISABLED_HDF5WriteADIOS2HDF5Read2D2x4)
 //******************************************************************************
 
 // ADIOS2 write, native HDF5 read
-TEST_F(HDF5WriteReadTest, ADIOS2HDF5WriteHDF5Read2D4x2)
+TEST_F(HDF5WriteReadTest, DISABLED_ADIOS2HDF5WriteHDF5Read2D4x2)
```
Comment (chuckatkins) [inline testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp]: Should this be disabled?

Comment (HaochengLIU) [reply testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp]: Nope. Fixed.

File: testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp (hunk 1 above)
Comment (williamfgc) [reply testing/adios2/engine/hdf5/TestHDF5WriteRead.cpp]: @chuckatkins we need this if we want to reuse the HDF5 common object inside an application specific Engine (something more complex) that supports several transports. Today's engines are not first-class citizens, they are test engines for the toolkit components, that's why the logic is lightweight.

**Comment by HaochengLIU** [comment]
@scottwittenburg 
Hi Scott, it seems that CI/appveyor is not happy with me. Any ideas?

**Comment by scottwittenburg** [comment]
At the moment I can't understand why appveyor is building/testing your branch, as so far it's something I have only set up on my own fork of the project.  Maybe @chuckatkins did some midnight work while I wasn't looking though?  At any rate, I'll look into it.

**Comment by scottwittenburg** [comment]
I have revoked all appveyors authorizations which I control.  Can you push again on this branch and see if it still tries to run that check?

**Comment by scottwittenburg** [comment]
Now I have found out how to log in to AppVeyor on behalf of the `ornladios` user, and then turn off a) building of feature branches which are part of a PR, and b) building of branches without an `appveyor.yml`, so I think you should be good to go for now.

Just doing an empty `git commit --amend` and then force pushing should be enough to check.

**Comment by scottwittenburg** [comment]
Awesome @HaochengLIU, looks like you're all set again.  Sorry for the disruption.

**Comment by HaochengLIU** [comment]
@scottwittenburg 
Hi Scott, I think your fix works. Thanks!

Btw, the **rebuild with SSH** action is grey in CI for me, saying that `You need write permission to trigger build`, I guess I need to be granted to some level?

**Comment by scottwittenburg** [comment]
Yes, now that you mention it, I had a similar issue until @chuckatkins granted some extra level of access for me.  I think I needed write access to the ornladios/adios2 repo.  And even then, it seems it didn't update that greyed out button right away.  I guess we'll need to look more deeply into how we can allow developers to rebuild with ssh without granting these write permissions.

Just to see what happens, I triggered a rebuild with ssh on this PR.  Can you see if you're able to go to the test results dashboard of CircleCI, get the ssh login info, and use it to login to the build?

**Comment by HaochengLIU** [comment]
I tried `ssh -i ~/.ssh/id_rsaGithub -p 64538 34.203.38.1677` with my github ssh key from the [cl website](https://circleci.com/gh/ornladios/ADIOS2/259?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) but it says that Permission denied(publickey). May I ask did I miss something here?

**Pull request merged**